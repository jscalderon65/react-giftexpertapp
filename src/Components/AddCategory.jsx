import React, { useState } from "react";
import PropTypes from "prop-types";
const AddCategory = ({setCategories}) => {
  const [inputValue, setInputValue] = useState("");
  return (
    <form onSubmit={(e)=>{
        e.preventDefault();
        if(inputValue.trim().length > 2) {
            setCategories(item=>[inputValue,...item]);
            setInputValue("")
        }
    }}>
      <hr />
      <input
        placeholder="Ingresa una categorÃ­a"
        type="text"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
        className="form-control"
      />
    </form>
  );
};
AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired
}
export default AddCategory;
