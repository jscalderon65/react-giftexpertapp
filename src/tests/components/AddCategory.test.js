import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import AddCategory from "../../Components/AddCategory";

describe("Pruebas en <GifGridItem />", () => {
  const setCategories = () => jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test("Debe mostrarse correctamente", () => {
    
    expect(wrapper).toMatchSnapshot();
  
});
  test("Debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";
    
    input.simulate("change", { target: { value } });
  });
  test("SimulaciÃ³n de limpiar caja de texto", () => {
    const value = "hola mundo";
    
    wrapper.find("input").simulate("change", { target: { value } });
    
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    expect(wrapper.find("input").prop("value")).toBe("");

});
});
