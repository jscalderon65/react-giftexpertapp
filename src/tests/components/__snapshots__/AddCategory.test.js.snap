// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <hr />
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Ingresa una categorÃ­a"
    type="text"
    value=""
  />
</form>
`;
